backend/app/services/llm_service.py

```
@@ -128,10 +123,160 @@ def _extract_json_from_response(self, response: str) -> Dict[str, Any]:
+        """
+        Initialize the Learning service.
+
+        Args:
+            config_service: The configuration service for persisting feedback.
+            max_feedback_size: Maximum number of feedback entries per domain.
+        """
+        self.config_service = config_service
+        self.max_feedback_size = max_feedback_size
+        self.feedback_data = self._load_feedback_data()
+
+    def _load_feedback_data(self) -> Dict[str, deque]:
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)